plugins/avro-format/core/src/test/java/org/pentaho/di/trans/steps/avro/input/PentahoAvroReadWriteTest.java

```diff
@@ -784,4 +784,16 @@ private ValueMetaInterface getValueMetaInterface( String fieldName, int fieldTyp
     return null;
   }
 
+  @Test
+  public void testAvroFileWriteAndReadDefaultValues() throws Exception {
+    Object[] rowData = new Object[] { "Row3Field1", null, null, null, null, null, null, null, null, null };
+
+    String[] defaultValues =
```

Comment:
Are you missing some  AvroSpec.DataType  cases in your test?

I'm unfamiliar with ` AvroSpec.DataType`, there are switch cases for `FLOAT, LONG, INTEGER, DECIMAL` for current set of values `"1234.0", "5.5",  "-33456"`  in your test, which corresponds to which case?

for switch cases `TIMESTAMP_MILLIS, DATE` for the current set of values "1980/01/01 00:00:00.000", "1980/01/01 00:00:00.000" (they are the same value) in your rets, which corresponds to which case?

for switch case `BYTES` what test values supports the test case?

for the default switch case, what test values supports that test case?

Reply:
Addressed the test case, Added the time stamp in milliseconds format and a binary value for bytes. All the switch cases are covered for Avro output , hence so specific case for default. Do let me know if it can be removed.